tools/PTE/pte-snapshot.js

```diff
@@ -0,0 +1,191 @@
+    logger.info('[queryBlockInfo] block height= %d', parseInt(blockchainInfo.height));
+
+    return blockchainInfo.height;
+}
+
+/**
+ * query block info callback
+ *
+ * @param {object} channel The channel object
+ * @param {object} client The client object
+ * @param {string} org The target organization name
+ * @param {string} cpf The connection profile contains the target org
+ * @param {int} freq The query info frequency in ms
+ * @returns {Promise<void>}
+ */
+async function queryBlockInfoCB(channel, client, org, cpf, freq) {
+    let blkheight = await queryBlockInfo(channel, client, org, cpf);
+    if ( parseInt(blkheight) > parseInt(snapshot.height) ) {
+        logger.info('[queryBlockInfoCB] block height: target= %d, current= %d', snapshot.height, blkheight);
+        process.exit();
+    }
+
+    // schedule next query
+    setTimeout(function() {
+        queryBlockInfoCB(channel, client, org, cpf, freq);
+    }, freq);
+}
+
+/**
+ * query block info Handler
+ *
+ * @param {object} channel The channel object
+ * @param {object} client The client object
+ * @param {string} org The target organization name
+ * @param {string} cpf The connection profile contains the target org
+ * @param {int} freq The query info frequency in ms
+ * @returns {Promise<void>}
+ */
+async function queryBlockInfoHandler(channel, client, org, cpf, freq) {
+
+    // get client key if clientauth
+    if (TLS == testUtil.TLSCLIENTAUTH) {
+        await testUtil.tlsEnroll(client, org, cpf);
+        logger.debug('[queryBlockInfoHandler] got user private key: org= %s', org);
+    }
+
+
+    hfc.setConfigSetting('key-value-store', 'fabric-common/lib/impl/FileKeyValueStore.js');
+    let cryptoSuite = hfc.newCryptoSuite();
+    cryptoSuite.setCryptoKeyStore(hfc.newCryptoKeyStore({ path: testUtil.storePathForOrg(Nid, org) }));
+    client.setCryptoSuite(cryptoSuite);
+
+    testUtil.assignChannelOrdererSubmitter(channel, client, org, cpPath, TLS);
+
+    let tgtPeers = [];
+    let tgtOrg = [];
+    tgtOrg.push(org);
+    tgtPeers[tgtOrg]=[];
+    tgtPeers[tgtOrg].push(snapshot.peerName);
+    if ( tgtPeers ) {
+        testUtil.assignChannelPeersSubmitter(cpList, channel, client, tgtPeers, TLS, cpPath, null, null, null, null, null);
+    }
+
+    return hfc.newDefaultKeyValueStore({
+        path: testUtil.storePathForOrg(org)
+    }).then(function (store) {
+        client.setStateStore(store);
+        return testUtil.getSubmitter(username, secret, client, true, Nid, org, cpf);
+    }).then((admin) => {
+        return channel.initialize();
+    }).then((success) => {
+        logger.info('[queryBlockInfoHandler] successfully initialize channel');
+        // execute query info
+        queryBlockInfoCB(channel, client, org, cpf, freq);
+    }).catch((err) => {
+        logger.error('[queryBlockInfoHandler] initialize channel failed');
+        logger.error(err.stack ? err.stack : err);
+        process.exit(1);
+    });
+
+}
```

Comment:
The chaining of promises isn't really idiomatic anymore, lets swap it our for `await`, I think it looks like this:
```
let kvs = await hfc.newDefaultKeyValueStore({
    path: testUtil.storePathForOrg(org)
})
await client.setStateStore(kvs);
await testUtil.getSubmitter(username, secret, client, true, Nid, org, cpf);
await channel.initialize();
await logger.info('[queryBlockInfoHandler] successfully initialize channel');
try {
    return await queryBlockInfoCB(channel, client, org, cpf, freq);
} catch(err) {
    logger.error('[queryBlockInfoHandler] initialize channel failed');
    logger.error(err.stack ? err.stack : err);
    process.exit(1);
}
```
I can't decide if there should be an `await` on the call to `queryBlockInfoCB` or not so you'll need to make that call

Reply:
ok